### PR TITLE
Pretty print and some fixes

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/register_function.jl [hunk 1]
```diff
@@ -48,7 +48,7 @@ macro register(expr, Ts = [Num, Symbolic, Real])
     push!(
           ex.args,
           quote
-              if $!($hasmethod($promote_symtype, $Tuple{$typeof($f), $Vararg}))
+              if $define_promotion
                   (::$typeof($promote_symtype))(::$typeof($f), args...) = $ret_type
```
Comment (YingboMa) [inline src/register_function.jl:52]: When I use the `hasmethod` check, the `promote_symtype` method doesn't seem to be defined only when I run `@register` in a file. Any ideas? @shashi

File: src/ModelingToolkit.jl [hunk 2]
```diff
@@ -152,13 +152,13 @@ end
...
 tosymbolic(a::Sym) = tovar(a)
 tosymbolic(a) = a
-@num_method Base.isless isless(tosymbolic(a), tosymbolic(b)) (Real,)
-@num_method Base.:(<) (tosymbolic(a) < tosymbolic(b)) (Real,)
-@num_method Base.:(<=) (tosymbolic(a) <= tosymbolic(b)) (Real,)
-@num_method Base.:(>) (tosymbolic(a) > tosymbolic(b)) (Real,)
-@num_method Base.:(>=) (tosymbolic(a) >= tosymbolic(b)) (Real,)
-@num_method Base.isequal isequal(tosymbolic(a), tosymbolic(b)) (AbstractFloat, Number, Symbolic)
-@num_method Base.:(==) tosymbolic(a) == tosymbolic(b) (AbstractFloat,Number)
+@num_method Base.isless  Num(isless(tosymbolic(a), tosymbolic(b))) (Real,)
```
Comment (YingboMa) [inline src/ModelingToolkit.jl:155]: We need these `Num`s to make sure the invariance `foo(::Num) -> ::Num` is satisfied.

Comment (shashi) [reply src/ModelingToolkit.jl:155]: Okay, what we had was I think what `Constant` did.

File: src/register_function.jl (hunk 1 above)
Comment (shashi) [reply src/register_function.jl:52]: I'm not sure

File: src/ModelingToolkit.jl (hunk 2 above)
Comment (YingboMa) [reply src/ModelingToolkit.jl]: I changed this to `val isa Bool ? val : Num(val)`, so that `Num(1) == 1` is true to be more user-friendly

**Comment by YingboMa** [comment]
Test failure is unrelated. All tests pass locally.

**Pull request merged**